### PR TITLE
network

### PR DESCRIPTION
**Pull request by diasdavid**
This is a **WIP**, missing: 

- [x] Update Swarm with `peer-mux-*` events
- [x] Update peerBook on js-ipfs to make it smarter when it comes to updating the info about a peer (with the option to flush it to disk) and indexing. So that I can fetch the peerInfo for `sendMessage` call
- [x] Update Swarm to patch the peerId to the conn (and the streams that are multiplexed on the conn), so that we can do the `_receiveMessage`

### CODE REVIEW

File: src/network/index.js [hunk 1]
```diff
@@ -1,14 +1,38 @@
...
   }
 
   // Connect to the given peer
   connectTo (peerId, cb) {
-    // TODO: Implement me
+    // NOTE: For now, all this does is ensure that we are
+    // connected. Once we have Peer Routing, we will be able
+    // to find the Peer
+    if (this.libp2p.swarm.muxedConns[peerId.toB58String()]) {
+      // TODO: ask - want _onPeerConnected to be called here too?
```
Comment (diasdavid) [inline src/network/index.js]: @dignifiedquire should I call _onPeerConnected here too? Or is the `cb` enough to signal we successfully  dialed (I'm afraid that we fire to routines for connecting to the same peer)

**Comment by dignifiedquire** [comment]
@diasdavid can you rebase please?

File: src/network/index.js (hunk 1 above)
Comment (dignifiedquire) [inline src/network/index.js]: I need to check

**Comment by diasdavid** [comment]
@dignifiedquire rebased, but do not merge yet :)

File: src/network/index.js (hunk 1 above)
Comment (dignifiedquire) [inline src/network/index.js]: No need to call `_onPeerConnected` here, as that will already have been called from the first connection in this case.

**Comment by diasdavid** [comment]
@dignifiedquire the network is implemented :)

File: test/index-test.js [hunk 2]
```diff
@@ -146,7 +150,10 @@ module.exports = (repo) => {
       })
     })
 
-    it('block is retrived from peer', (done) => {
+    // Not sure if I understand what is going on here
+    // test fails because now the network is not properly mocked
+    // what are these net.stores and mockNet.bitswaps?
```
Comment (diasdavid) [inline test/index-test.js]: @dignifiedquire could you clarify what is going on in this test? libp2p needs to be properly mocked, but I couldn't get what was exactly was going on/intended.

Comment (dignifiedquire) [inline test/index-test.js]: Yes, I need to change the mock network tool to now mock libp2p instead of the network instance

**Pull request merged**